mlflow/data/dataset_source.py

```diff
@@ -5,6 +5,19 @@
 from mlflow.utils.annotations import experimental
 
 
+class _JsonEncoderStr(json.JSONEncoder):
+    """
+    A custom JSON encoder that falls back to string representation for objects that cannot be
+    serialized by the default JSON encoder.
+    """
+
+    def default(self, obj):
+        try:
+            return super().default(obj)
+        except TypeError:
```

Comment:
Should we log a warning (info level) if this is being done so that the user is alerted to the fact that they won't be able to use standard deserialization through `json.loads()` for data processed by this custom serializer? 
(Not a blocker)